brainscore/benchmarks/zhu2019.py

```diff
@@ -0,0 +1,207 @@
+            identifier='Zhu2019_extreme_occlusion-accuracy',
+            parent='Zhu2019',
+            ceiling_func=lambda: Score([1, np.nan], coords={'aggregation': ['center', 'error']}, dims=['aggregation']),
+            bibtex=BIBTEX, version=1)
+
+    def __call__(self, candidate: BrainModel):
+
+        # hard code categories, as sorting alphabetically leads to mismatched IDs
+        categories = ["car", "aeroplane", "motorbike", "bicycle", "bus"]
+        candidate.start_task(BrainModel.Task.label, categories)
+        stimulus_set = place_on_screen(self._assembly.stimulus_set, target_visual_degrees=candidate.visual_degrees(),
+                                       source_visual_degrees=self._visual_degrees)
+        predictions = candidate.look_at(stimulus_set, number_of_trials=self._number_of_trials)
+        predictions = predictions.sortby("stimulus_id")
+
+        # grab ground_truth from predictions (linked via stimulus_id) instead of stimulus set, to ensure sort
+        ground_truth = predictions["ground_truth"].sortby("stimulus_id")
+
+        # compare model with stimulus_set (ground_truth)
+        raw_score = self._metric(predictions, ground_truth)
+        ceiling = self.ceiling
+        score = raw_score / ceiling.sel(aggregation='center')
+
+        # cap score at 1 if ceiled score > 1
+        if score[(score['aggregation'] == 'center')] > 1:
+            score.__setitem__({'aggregation': score['aggregation'] == 'center'}, 1)
+
+        score.attrs['raw'] = raw_score
+        score.attrs['ceiling'] = ceiling
+        return score
+
+
+def _human_assembly_categorical_distribution(assembly: DataAssembly, collapse) -> DataAssembly:
+    """
+    Convert from 19587 trials across 25 subjects to a 500 images x 5 choices assembly.
+    This is needed, as not every subject saw every image, and this allows a cross-category comparison
+    """
+
+    categories = ["car", "aeroplane", "motorbike", "bicycle", "bus"]
+
+    def categorical(image_responses):
+        frequency = np.array([sum(image_responses.values == category) for category in categories])
+        frequency = frequency / len(image_responses)
+        frequency = type(image_responses)(frequency, coords={'choice': categories}, dims=['choice'])
+        return frequency
+
+    stimulus_coords = ['stimulus_id', 'truth', 'filename', 'image_label', 'ground_truth',
+                       'occlusion_strength', 'image_number', 'word_image']
+    categorical_assembly = assembly.multi_groupby(stimulus_coords).map(categorical)
+
+    # get top average human response, across categories per image.
+    labels = get_choices(categorical_assembly, categories=categories)
+
+    if collapse:
+        categorical_assembly = categorical_assembly.groupby('image_label').sum('presentation')
+        return categorical_assembly
+
+    return labels
+
+
+# takes 5-way softmax vector and returns category of highest response
+def get_choices(predictions, categories):
```

Comment:
if `predictions` is an assembly, you shouldn't need this function or at least not as much code logic. xarray has an [`argmax`](https://docs.xarray.dev/en/stable/generated/xarray.DataArray.argmax.html) function that you can use rather than going into the raw numpy values.

Reply:
Currently using argmax for this - pulled straight from LabelBehavior in behavior.py